src/phases/execution/MultiDeletes.yml

```diff
@@ -0,0 +1,163 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/server-execution"
+Description: >
+  Runs inserts on one collection while a mass deletion takes place on another collection in the same database.
+
+GlobalDefaults:
+  # Configurable parameters
+  DocumentSizePaddingBytes: &DocumentSizePaddingBytes {^Parameter: {Name: "DocumentSizePaddingBytes", Default: 0}}
+  DocumentCount: &DocumentCount {^Parameter: {Name: "DocumentCount", Default: 100_000_000}}
+
+# Static variables.
+WriteRate: &WriteRate 1 per 500 microseconds  # 2000/second
+Database: &Database test
+# Collection0 is the default collection for the MonotonicSingleLoader.
+MassDeletionsColl: &MassDeletionsColl Collection0
+
+Actors:
+
+# Phase 1: Drop the database.
+- Name: DropDatabase
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: *Database
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand: {dropDatabase: 1}
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+  - *Nop
+
+# Phase 2: Populate the collection for mass deletion.
+- Name: PopulateInitialData0
+  Type: MonotonicSingleLoader
+  Threads: 100
+  Phases:
+  - *Nop
+  - Repeat: 1
+    MetricsName: PrePopulation0
+    Database: *Database
+    Collection: *MassDeletionsColl
+    DocumentCount: *DocumentCount
+    BatchSize: 1000
+    Document:
+      padding: {^FastRandomString: {length: *DocumentSizePaddingBytes}}
+  - *Nop
+  - *Nop
+  - *Nop
+
+# Phase 2: Gaurd against timout due to no logging during PrePoulation0.
+- Name: LoggingActor0
+  Type: LoggingActor
+  Threads: 1
+  Phases:
+  - *Nop
+  - LogEvery: 9 minutes
+    Blocking: None
+  - *Nop
+  - *Nop
+  - *Nop
+
+# Phase 3: Issue a large delete that should take a considerable amount of time to run. This delete
+# will run with no concurrent writes to the database.
+- Name: LargeMultiDelete0
+  Type: CrudActor
+  Database: *Database
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    MetricsName: DuringMassDeletion0
+    Collection: *MassDeletionsColl
+    Operations:
+    - OperationName: bulkWrite
+      OperationCommand:
+        WriteOperations:
+        - WriteCommand: deleteMany
+          Filter: {_id: {$gte: 0}}
+          OperationOptions:
+            WriteConcern: { Level: "majority" }
+  - *Nop
+  - *Nop
+
+# Phase 4: Populate the collection for mass deletion again. This time concurrent writes will be
+# issued to another collection in the same database.
+# NOTE: The MonotonicSingleLoader counter will start where it left off in the first instance of this
+# Actor in a previous phase.
+- Name: PopulateInitialData1
```

Comment:
I'm seeing a problem while running it locally. Phase 1 through phase 3 completed. However phase 4 doesn't seem to run, and phase 5 seems to have run for a tiny fraction of time, since I found just 1 document (and only 7 documents in another attempt) inserted in `test.Collection1`. Do you see the same on your end?

Reply:
Phase 5 runs as long as it takes to delete all the documents. The inserts into Collection1 only run for a fraction of time since the delete didn't have much work to do. It appears the issue is with phase 4

Reply:
Looks resolved in the current revision. Good job!